core/trino-main/src/main/java/io/trino/operator/exchange/UniformPartitionRebalancer.java

```diff
@@ -162,17 +162,15 @@ private synchronized void rebalancePartitions(List<Long> writerPhysicalWrittenBy
             for (WriterId minSkewedWriter : minSkewedWriters) {
                 // There's no need to add the maxWriter back to priority queues if no partition rebalancing happened
                 if (context.rebalancePartition(maxWriter, minSkewedWriter)) {
-                    maxWriters.addOrUpdate(maxWriter, context.getWriterEstimatedWrittenBytes(maxWriter));
-                    minWriters.addOrUpdate(maxWriter, Long.MAX_VALUE - context.getWriterEstimatedWrittenBytes(maxWriter));
+                    for (Integer writer: context.rebalancedAffectedWriters) {
+                        WriterId writerId = new WriterId(writer);
+                        maxWriters.addOrUpdate(writerId, context.getWriterEstimatedWrittenBytes(maxWriter));
+                        minWriters.addOrUpdate(writerId, Long.MAX_VALUE - context.getWriterEstimatedWrittenBytes(maxWriter));
+                    }
                     break;
                 }
             }
 
-            // Add all the min skewed writers back to the minWriters queue with updated priorities
-            for (WriterId minSkewedWriter : minSkewedWriters) {
```

Comment:
I think this is still needed. `findSkewedMinWriters` will pool min writers, but `rebalancePartition` might not rebalance anything (e.g. because partition is already written by multiple writers). Hence, `min skewed` writers need to be put back into queue for next `maxWriter`